Snakefile

```diff
@@ -1,11 +1,34 @@
-configfile: "config/config_dengue.yaml"
+from snakemake.utils import min_version
+min_version("6.0")
+
+# Use default pathogen build config if no configs are provided
+if not config:
+    configfile: "config/config_dengue.yaml"
+# Use default ingest config if no `transform` config is provided
+if not config.get("transform"):
+    configfile: "ingest/config/config.yaml"
+
+# Add the hard-coded ingest basedir to the workflow config so that we can
+# pass it to the module ingest workflow. This will allow shell scripts to
+# use the proper paths for local script invocation since we cannot set the
+# workdir separately for module workflows.
+# This work around is based on https://stackoverflow.com/a/66890412
+config["ingest_basedir"] = f"{workflow.current_basedir}/ingest"
 
 serotypes = ['all', 'denv1', 'denv2', 'denv3', 'denv4']
 
 rule all:
     input:
         auspice_json = expand("auspice/dengue_{serotype}.json", serotype=serotypes)
 
+module ingest_workflow:
```

Comment:
I haven't used modules myself, so I'm interested to know how you and @joverlee521 found them to work with - especially related to debugging / developing? 

As a counterexample, I've been playing with [an HBV build](https://github.com/nextstrain/hepatitisB/blob/master/Snakefile) which uses our convention of separating out ingest data and code into `./ingest/`, but still runs everything from a top-level `./Snakemake` (which calls `include: "ingest/ingest.smk"`). 

The advantage of modules (as used in this context) is that they continue to allow `./ingest/` to function in a completely stand-alone way, which has obvious advantages. Perhaps using modules also enforces the separation of concerns between ingest + phylo in a desirable way? The cost is increased complexity. After seeing how complex the `ncov` snakemake workflow became over the years I'm trying to gauge whether the trade-offs here are worth it.